test/libsolidity/semanticTests/array/copying/dirty_memory_bytes_to_storage_copy.sol

```diff
@@ -16,4 +16,4 @@ contract C {
 // ====
 // compileViaYul: false
 // ----
-// f() -> 0x6465616462656566313564656164000000000000000000000000000000000010
+// f() -> 0x6465616462656566000000000000000000000000000000000000000000000010
```

Comment:
This test means we were fine with this in the past... did we realize back than that the dirty values can actually be accessed without inline assembly when using ``.push()`` on the array?
Given that are we *still* fine with this?
We could also just disable empty ``.push()`` on bytes instead... or have to actually write a zero value...

Reply:
> did we realize back than that the dirty values can actually be accessed without inline assembly when using .push() on the array?

No, not as far as I can recall.

Reply:
I think this was overlooked. But it seems like nobody cared anyway

Reply:
I don't think empty ``push``es to bytes are really *that* rare :-). Merely ones that don't write anything eventually anyways... and actually writing zeroes is a huge hit for those cases, unless the write can be optimized away...
And it'd be weird to write zeroes here all of a sudden, but in no other cases...

Reply:
But I thought we write zeros in all other cases? If we don't then why is it a problem only with `bytes` arrays.

Reply:
We never write zeroes :-). Storage is just always zeroed. For packed arrays writing an explicit zero is an ``sload`` and an ``store`` for each element (or at least each slot, but only if the optimizer works well on it) - that's among the most costly things you can do :-).

What we do is that we write zeroes on *deallocation*, i.e. on ``delete`` and ``.pop()`` and the like, s.t. we *can* assume storage to always be zero.

The bug is that copies from calldata/memory to storage do *not* respect the general principle that storage never contains dirty values, thereby violating the assumption that all unused storage is zero at all times.